File: backend/src/main/java/dev/anthonybruno/gymbuddy/util/ClassPathFile.java
package dev.anthonybruno.gymbuddy.util;

import java.io.InputStream;
import java.net.URL;

public class ClassPathFile {

    private final Class<?> context;
    private final String location;

    public ClassPathFile(String location) {
        this(location, ClassPathFile.class);
    }

    public ClassPathFile(String location, Class<?> context) {
        this.location = location;
        this.context = context;
    }

    public boolean exists() {
        return asStream() != null;
    }

    public InputStream asStream() {
        return getClassLoader().getResourceAsStream(location);
    }

    public URL asUrl() {
        return getClassLoader().getResource(location);
    }

    private ClassLoader getClassLoader() {
        return context.getClassLoader();
    }


}

File: backend/src/main/kotlin/dev/anthonybruno/gymbuddy/util/ClassPathFile.kt
package dev.anthonybruno.gymbuddy.util

import java.io.InputStream
import java.net.URISyntaxException
import java.net.URL
import java.nio.file.Path
import java.nio.file.Paths

class ClassPathFile(private val location: String, private val context: Class<*> = ClassPathFile::class.java) {

    private val classLoader: ClassLoader
        get() = context.classLoader

    fun exists(): Boolean {
        return asStream() != null
    }

    fun asStream(): InputStream? {
        return classLoader.getResourceAsStream(location)
    }

    fun asUrl(): URL? {
        return classLoader.getResource(location)
    }

    fun asPath(): Path {
        try {
            return Paths.get(asUrl()!!.toURI())
        } catch (e: URISyntaxException) {
            throw RuntimeException(e)
        }

    }


}
